### PR TITLE
feat: Configurable url context parameters (eg. currency, site, lang)

### PR DESCRIPTION
**Pull request by dunqan**

### CODE REVIEW

File: projects/core/src/site-context/services/site-context-url-serializer.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        params[paramName] = segments[i];
+      } else {
+        break;
+      }
+    }
+    url = segments.slice(Object.keys(params).length).join('/');
+    return { url, params };
+  }
+
+  private urlTreeIncludeCotextParameters(
```
Comment (Platonn) [inline projects/core/src/site-context/services/site-context-url-serializer.ts]: typo `Cotext` -> `Context`

File: projects/core/src/site-context/facade/site-context-params.service.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          key => contextConfig[key].persistence === persistence
+        );
+      } else {
+        return params;
+      }
+    }
+    return [];
+  }
+
+  getParamValues(param: string) {
```
[the remark this answers is not in the document]
Comment (dunqan) [reply projects/core/src/site-context/facade/site-context-params.service.ts]: 👍

File: projects/core/src/site-context/facade/site-context-params.service.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  constructor(
+    private config: SiteContextConfig,
+    private injector: Injector,
+    private serviceMap: ContextServiceMap
+  ) {}
+
+  getContextParameters(persistence?: ContextParamPersistence): string[] {
+    const contextConfig = this.config.siteContext;
+    if (contextConfig) {
+      const params = Object.keys(contextConfig);
```
[the remark this answers is not in the document]
Comment (dunqan) [reply projects/core/src/site-context/facade/site-context-params.service.ts]: Thakns for catching this!

File: projects/core/src/site-context/services/site-context-url-serializer.ts (hunk 1 above)
Comment (dunqan) [reply projects/core/src/site-context/services/site-context-url-serializer.ts]: 👍

File: projects/core/src/site-context/services/site-context-url-serializer.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      let i = 0;
+      i < this.routeContextParameters.length && i < segments.length;
+      i++
+    ) {
+      const paramName = this.routeContextParameters[i];
+      const paramValues = this.siteContextParams.getParamValues(paramName);
+      if (paramValues.indexOf(segments[i]) > -1) {
+        params[paramName] = segments[i];
+      } else {
+        break;
```
[the remark this answers is not in the document]
Comment (dunqan) [reply projects/core/src/site-context/services/site-context-url-serializer.ts]: Yup, this is by spec (this is how it also works on the backend) so the description should go to docs directly.

**Pull request merged**